Neos.Media.Browser/Classes/Domain/Session/BrowserState.php

```diff
@@ -40,6 +48,10 @@ class BrowserState
      */
     public function set(string $key, $value)
     {
+        if (is_object($value) && $entityIdentifier = $this->persistenceManager->getIdentifierByObject($value)) {
```

Comment:
hey @dlubitz, I don't get completely. The assetCollection is set in AssetController L168, which contains the asset objects. So this would not fix the issue with the assets in that case?

Reply:
It does, as it just stores the identifier and class name in "data" of the browser state, instead of the full entitiy.